pkg/apis/gcp/v1alpha1/types_infrastructure.go

```diff
@@ -105,12 +109,27 @@ type CloudRouter struct {
 	Name string `json:"name,omitempty"`
 }
 
-// CloudNAT contains information about the the CloudNAT configuration
+// CloudNAT contains configuration about the the CloudNAT resource
 type CloudNAT struct {
 	// MinPortsPerVM is the minimum number of ports allocated to a VM in the NAT config.
 	// The default value is 2048 ports.
 	// +optional
 	MinPortsPerVM *int32 `json:"minPortsPerVM,omitempty"`
+	// NatIPNames is a list of all user provided external premium ips which can be used by the nat gateway
+	// +optional
+	NatIPNames []NatIPName `json:"natIPNames,omitempty"`
+}
+
+// NatIP is a user provided external ip which can be used by the nat gateway
+type NatIP struct {
+	// Name of the external premium ip address which is used in gcp
```

Comment:
this is not the name anymore right? 
```suggestion
	// IP is the external premium IP address used in GCP
```